File: 0724-find-pivot-index/0724-find-pivot-index.c
/**
 * @brief Finds the pivot index in the array where the sum of elements to the left
 * is equal to the sum of elements to the right.
 *
 * @param nums (int*): Pointer to the array of integers.
 * @param numsSize (int): The size of the array.
 *
 * @return (int): The pivot index, or -1 if no such index exists.
 */
int pivotIndex(int* nums, int numsSize) {
    // Variable to store the total sum of the array
    int total_sum = 0;

    // Calculate the total sum of the elements in the array
    for (int i = 0; i < numsSize; i++) {
        total_sum += nums[i];
    }

    // Variable to store the sum of the left side of the array
    int left_sum = 0;

    // Iterate through the array to find the pivot index
    for (int i = 0; i < numsSize; i++) {
        // If the sum of the left side is equal to the sum of the right side
        // Right sum is total_sum - left_sum - nums[i]
        if (left_sum == total_sum - left_sum - nums[i]) {
            return i;  // Return the pivot index
        }
        // Update the left sum to include the current element
        left_sum += nums[i];
    }

    // If no pivot index is found, return -1
    return -1;
}

File: Explore Card Track/4. Array and String/0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        """
        Finds the pivot index in the list `nums` where the sum of elements to the left
        is equal to the sum of elements to the right. Returns -1 if no such index exists.

        Parameters:
        nums (List[int]): A list of integers.

        Returns:
        int: The pivot index or -1 if no such index exists.
        """
        # Calculate the total sum of the array
        total_sum = sum(nums)
        left_sum = 0

        # Iterate through the array to find the pivot index
        for i, num in enumerate(nums):
            # Check if the left sum equals the right sum (right sum is total_sum - left_sum - current num)
            if left_sum == total_sum - left_sum - num:
                return i
            # Update the left sum to include the current number
            left_sum += num

        # If no pivot index is found, return -1
        return -1
